### ISSUE TITLE
scrollbar diagnostics not working properly

### ISSUE THREAD

**Issue by sourabhrathour45** [zed-industries/zed#6819]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The scrollbar diagnostics are not accurate even if I change a single line, a yellow colour distracting thing appears on the whole scrollbar which is very annoying while coding.

### Environment

Zed: v0.119.19 (Zed)
OS: macOS 14.2.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/zed-industries/zed/assets/84420690/c0d83f1e-3624-46a8-b9e9-cf71a0929430)

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by JosephTLyons** [comment]
Hey @sourabhrathour45, these are not diagnostics in the scrollbar, these are git markers - they coincide with what you see in the gutter on the left side of the editor. In the future, i think we will have error diagnostics in the scrollbar as well. If you wish to turn off the git markers, you can configure Zed with:

`"git_diff": false`

**Comment by sourabhrathour45** [comment]
Hey @JosephTLyons, no actually i have hidden the git gutter in the left side of the editor by `"git-gutter": "hide"` , what i'm facing is these **inconsistent yellow markers** on the scrollbar of the editor, even if i make a **single line change** after cloning a project, the whole scrollbar turns yellow. 

Here's one example of the same - 
I have disabled the git gutter on the left side of the editor, thus it's not visible, but the the scrollbar is all flashy that's quite distracting to eyes. I also tried `"git-diff": false` but it didn't worked.

![image](https://github.com/zed-industries/zed/assets/84420690/c518080c-6dfe-4ee3-b6ec-71fdb1dc76f8)
